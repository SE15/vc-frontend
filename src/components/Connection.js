import {  Heading, Flex, Avatar, Box, HStack ,Link} from "@chakra-ui/react"

function Connection() {


    return (
        <>
            <Flex>
                <Box w="20rem" p={4} bg="white.200">
                    <HStack  spacing="3rem">
                    <Avatar name="Oshigaki Kisame" src="https://bit.ly/broken-link" />
                    <Heading size="md" color="black.600">Full Name</Heading>
                    <Link color="blue.200" fontSize="md"href="#">
                        View
                    </Link>
            
                    </HStack>
                </Box>
            </Flex>

        </>
    )
}

export default Connection;
