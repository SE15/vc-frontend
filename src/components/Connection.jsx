import {React,useState} from "react"
//import { Link } from "react-router-dom"
import PropTypes from "prop-types"

import {  Heading, Flex, Avatar, Box, HStack,Image,Link } from "@chakra-ui/react"
import { kPrimaryBlackDark, kPrimaryGray }  from './../utils/constants'

function Connections(props){

    const [image, setImage] = useState(props.cimage);
    return(
        <>
        <Flex>
            <Box w="100%" p={4} bg="{kPrimaryGray}">
                <HStack  spacing="3rem">
                {props.cimage!=null?(
                     <Image src={props.cimage}  borderRadius="full" boxSize="3rem" />
                 
                 ) :(
                     <Avatar name={props.cname} src="https://bit.ly/broken-link" />
                 )
                }
                
                <Heading size="sm" color={kPrimaryBlackDark}>{props.cname}</Heading>
                <Link color="blue.200" fontSize="md"href={props.plink} textAlign="end">
                        View
                    </Link>
        
                </HStack>
            </Box>
        </Flex>

    </>
    )
}
Connections.propTypes = {
    name: PropTypes.string,
    image: PropTypes.string,
    plink: PropTypes.string,
  }
  Connections.defaultProps =   {
      image:"null"
  }
  export default Connections  