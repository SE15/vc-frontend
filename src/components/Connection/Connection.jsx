import { React } from "react"
//import { Link } from "react-router-dom"
import PropTypes from "prop-types"

import { Text, Avatar, HStack, Button, Spacer } from "@chakra-ui/react"
import { ExternalLinkIcon } from '@chakra-ui/icons'
import { useHistory } from 'react-router-dom';

function Connection({ name, user, image, widthAuto, authUser }) {
    const history = useHistory();

    const viewProfile = () => {
        if (authUser) {
            if (authUser != user)
                history.push(`/profiles/${name}`, { user });
            else
                history.push('/');
        } else
            history.push(`/profiles/${name}`, { user });
    }

    return (
        <HStack
            overflowY="hidden"
            h="70px"
            minH={70}
            borderWidth="1px"
            borderRadius="lg"
            px={4}
            py={2}
            bg="purple.100"
            boxShadow="lg"
            spacing={4}
            width={widthAuto ? "100%" : "350px"}>
            <Avatar name={name} src={image} borderColor="purple.500"
                    showBorder borderWidth={1}/>
            <Text color="gray.700" fontWeight={500} align="center" fontSize={16} overflow="hidden" isTruncated>{name}</Text>
            <Spacer />
            <Button
                leftIcon={<ExternalLinkIcon />}
                variant="link"
                colorScheme="blue"
                pb={1}
                onClick={viewProfile}>
                view
            </Button>
        </HStack>
    )
}

export default Connection;