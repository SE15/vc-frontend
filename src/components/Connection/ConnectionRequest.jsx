import {
    Avatar,
    Button,
    HStack,
    Spacer,
    StackDivider,
    VStack,
} from "@chakra-ui/react"
import {
    CheckIcon,
    CloseIcon,
} from '@chakra-ui/icons'
import { useHistory } from 'react-router-dom';

const ConnectionRequest = ({ name, onAccept, onReject, user, loader, loadingId }) => {
    const history = useHistory();

    return (
        <HStack borderWidth="1px" borderRadius="lg" bg="white" px={4} py={2} bg="purple.100" boxShadow="lg">
            <Avatar name={name} />
            <VStack spacing={1}>
                <Button
                    fontSize="16px"
                    color="gray.700"
                    onClick={() => { history.push(`/profiles/${name}`, { user }) }}
                    variant="link"
                >
                    {name}
                </Button>
                <StackDivider borderWidth="1px" borderColor="purple.200" />
                <HStack>
                    <Button
                        leftIcon={<CheckIcon boxSize="12px" pb="2px" />}
                        size="sm"
                        variant="ghost"
                        colorScheme="blue"
                        onClick={onAccept}
                        w="90px"
                        isLoading={loader===1 && loadingId===user}
                        isDisabled={loader===2 && loadingId===user}
                    >
                        Accept
                    </Button>
                    <Spacer />
                    <Button
                        leftIcon={<CloseIcon boxSize="12px" pb={1} />}
                        size="sm"
                        variant="ghost"
                        colorScheme="red"
                        onClick={onReject}
                        w="90px"
                        isLoading={loader===2 && loadingId===user}
                        isDisabled={loader===1 && loadingId===user}
                    >
                        Reject
                    </Button>
                </HStack>
            </VStack>
        </HStack>
    );
}

export default ConnectionRequest