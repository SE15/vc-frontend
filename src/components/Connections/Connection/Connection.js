import React from "react"
//import { Link } from "react-router-dom"
import PropTypes from "prop-types"

import {  Heading, Flex, Avatar, Box, HStack,Image,Link } from "@chakra-ui/react"

const Connections=props=>{


    return(
        <>
        <Flex>
            <Box w="28rem" p={4} bg="white.200">
                <HStack  spacing="3rem">
                 if({props.image}!=null){
                     <Image src={props.image} size="100%" rounded="1rem" shadow="2xl" />
                 }
                 else{
                     <Avatar name={props.name} src="https://bit.ly/broken-link" />
                 }   
                
                <Heading size="md" color="black.600">{props.name}</Heading>
                <Link color="blue.200" fontSize="md"href={props.plink}>
                        View
                    </Link>
        
                </HStack>
            </Box>
        </Flex>

    </>
    )
}
Connections.propTypes = {
    name: PropTypes.string,
    image: PropTypes.string,
    plink: PropTypes.string,
  }
  Connections.defaultProps =   {
      image:"null"
  }
  export default Connections  