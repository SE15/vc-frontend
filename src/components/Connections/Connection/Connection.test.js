import React from 'react';

import {configure,render,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { screen } from '@testing-library/dom';
import Connection from './Connection';

configure({adapter:new Adapter()});

describe('<Connection />',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper =shallow(<Connection cname="Anne De Mel" plink="https://www.test.com/"/>);
    });

    it('should match snapshot with correct name, image, link',()=>{
        expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('should match snapshot with correct name, image, link for different connection name',()=>{
        wrapper.setProps({cname:"Emma Watson",plink:"https://www.emma.com/"});
        expect(wrapper.getElements()).toMatchSnapshot();
    });



});    