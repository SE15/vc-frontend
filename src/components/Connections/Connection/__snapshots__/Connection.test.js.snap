// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Connection /> should match snapshot with correct name, image, link 1`] = `
Array [
  <React.Fragment>
    <Flex>
      <Box
        bg="#E2E8F0"
        p={4}
        w="390px"
      >
        <HStack
          spacing="3rem"
        >
          <Avatar
            name="Anne De Mel"
            src="https://bit.ly/broken-link"
          />
          <Heading
            color="#000a12"
            size="sm"
          >
            Anne De Mel
          </Heading>
          <Link
            color="blue.200"
            fontSize="md"
            href="https://www.test.com/"
            textAlign="end"
          >
            View
          </Link>
        </HStack>
      </Box>
    </Flex>
  </React.Fragment>,
]
`;

exports[`<Connection /> should match snapshot with correct name, image, link for different connection name 1`] = `
Array [
  <React.Fragment>
    <Flex>
      <Box
        bg="#E2E8F0"
        p={4}
        w="390px"
      >
        <HStack
          spacing="3rem"
        >
          <Avatar
            name="Emma Watson"
            src="https://bit.ly/broken-link"
          />
          <Heading
            color="#000a12"
            size="sm"
          >
            Emma Watson
          </Heading>
          <Link
            color="blue.200"
            fontSize="md"
            href="https://www.emma.com/"
            textAlign="end"
          >
            View
          </Link>
        </HStack>
      </Box>
    </Flex>
  </React.Fragment>,
]
`;
